### ISSUE TITLE
[dynamo] Data-dependent output shape ops (nonzero, repeat_interleave, masked_select) (65 models, 670 breaks)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#18]
## Summary

65 models produce 670 graph breaks from this pattern.

## Break Reason

```
Graph Break Reason: Failed to handle graph break gracefully. Skipping the function and falling back to eager. Graph break encountered:
```

## Affected Models

| Model | Modes | Breaks |
|-------|-------|--------|
| AriaForConditionalGeneration | eval, train | 10 |
| AriaModel | eval, train | 10 |
| AudioFlamingo3ForConditionalGeneration | eval, train | 5 |
| AyaVisionForConditionalGeneration | eval, train | 4 |
| Cohere2VisionForConditionalGeneration | eval, train | 10 |
| DbrxForCausalLM | eval, train | 20 |
| DbrxModel | eval, train | 16 |
| DeepseekVLForConditionalGeneration | eval, train | 10 |
| Ernie4_5_VLMoeForConditionalGeneration | eval, train | 12 |
| Ernie4_5_VL_MoeForConditionalGeneration | eval, train | 12 |
| FastSpeech2ConformerModel | eval, train | 12 |
| Florence2ForConditionalGeneration | eval, train | 10 |
| Florence2Model | eval, train | 6 |
| Gemma3ForConditionalGeneration | eval, train | 10 |
| Gemma4-vision | eval | 10 |
| Gemma4ForConditionalGeneration | eval, train | 20 |
| Glm46VForConditionalGeneration | eval, train | 12 |
| Glm4vForConditionalGeneration | eval, train | 12 |
| Glm4vMoeForConditionalGeneration | eval, train | 12 |
| Glm4vMoeVisionModel | eval, train | 2 |
| Glm4vVisionModel | eval, train | 2 |
| GlmImageVisionModel | eval, train | 2 |
| GlmOcrForConditionalGeneration | eval, train | 12 |
| GlmOcrVisionModel | eval, train | 2 |
| GotOcr2ForConditionalGeneration | eval, train | 10 |
| GotOcr2Model | eval, train | 6 |
| InternVLForConditionalGeneration | eval, train | 10 |
| InternVLModel | eval, train | 6 |
| JanusForConditionalGeneration | eval, train | 10 |
| Lfm2VlForConditionalGeneration | eval, train | 4 |
| LightOnOcrForConditionalGeneration | eval, train | 4 |
| Llama4ForConditionalGeneration | eval, train | 8 |
| LlavaForConditionalGeneration | eval, train | 4 |
| LlavaNextForConditionalGeneration | eval, train | 4 |
| LlavaNextVideoForConditionalGeneration | eval, train | 4 |
| LlavaOnevisionForConditionalGeneration | eval, train | 4 |
| LongcatFlashForCausalLM | eval, train | 20 |
| LongcatFlashModel | eval, train | 16 |
| MiniCPM-V-ViT | eval, train | 8 |
| Mistral3ForConditionalGeneration | eval, train | 4 |
| OpenAIPrivacyFilterModel | eval, train | 16 |
| OpenVoice-SynthesizerTrn-infer | eval | 8 |
| Ovis2ForConditionalGeneration | eval, train | 10 |
| PaddleOCRVLForConditionalGeneration | eval, train | 22 |
| PaliGemmaForConditionalGeneration | eval, train | 10 |
| PaliGemmaModel | eval, train | 6 |
| Phi4MultimodalAudioModel | eval, train | 6 |
| Qwen2AudioForConditionalGeneration | eval, train | 3 |
| Qwen2VLForConditionalGeneration | eval, train | 12 |
| Qwen2_5OmniThinkerForConditionalGeneration | eval, train | 14 |
| Qwen2_5_VLForConditionalGeneration | eval, train | 16 |
| Qwen3OmniMoeThinkerForConditionalGeneration | eval, train | 16 |
| Qwen3VLForConditionalGeneration | eval, train | 18 |
| Qwen3VLMoeForConditionalGeneration | eval, train | 18 |
| Qwen3_5ForConditionalGeneration | eval, train | 12 |
| Qwen3_5MoeForConditionalGeneration | eval, train | 12 |
| SmolVLMForConditionalGeneration | eval, train | 30 |
| SwitchTransformersEncoderModel | eval, train | 12 |
| SwitchTransformersForConditionalGeneration | eval, train | 16 |
| SwitchTransformersModel | eval, train | 16 |
| VideoLlama3ForConditionalGeneration | eval, train | 6 |
| VideoLlama3VisionModel | eval, train | 2 |
| ViltModel | eval, train | 10 |
| VipLlavaForConditionalGeneration | eval, train | 4 |
| VitsModel | eval, train | 20 |

## Impact

Fixing this pattern moves **2 models** to fullgraph (models where this is the only break reason).

## Related Issues

- #2
- #8
- #10
- #11
- #12
- #14
- #16
- #17
- #23
- #24
- #54
- #55
- #56

## Tested On

- PyTorch 2.13.0.dev20260425+cu126
- Transformers 5.6.2
- Sweep date: 2026-04-26T08:48:25

**Comment by penguinwu** [comment]
### Nightly Validation (torch 2.12.0.dev20260407 vs 2.11.0 stable)

**Status: Persists.** All 16 affected models still `graph_break` on both v2.11 and nightly (eval and train). `aten.nonzero`, `repeat_interleave`, and `masked_select` remain untraceable.

**Conclusion: No change between v2.11 and nightly.**

**Comment by penguinwu** [comment]
## Sweep update — 2026-04-28

Verified on: `torch 2.13.0a0+gitf8d66d2` (source build, includes PR #181552) · transformers `5.6.2` · `fullgraph=True, dynamic=True`

**Status: Partial progress — 1/62 models pass fullgraph, 44 still breaking**

**Now passing fullgraph:**
- ✅ `SmolVLMForConditionalGeneration`

**Changed status (infra error, not graph-break):** `Ernie4_5_VLMoeForConditionalGeneration`, `Ernie4_5_VL_MoeForConditionalGeneration`, `Glm46VForConditionalGeneration`, `Glm4vForConditionalGeneration`, `Glm4vMoeForConditionalGeneration`, `GlmOcrForConditionalGeneration`, `Mistral3ForConditionalGeneration`, `PaddleOCRVLForConditionalGeneration`, `Qwen2VLForConditionalGeneration`, `Qwen2_5_VLForConditionalGeneration`, `Qwen3VLForConditionalGeneration`, `Qwen3VLMoeForConditionalGeneration`, `Qwen3_5ForConditionalGeneration`, `Qwen3_5MoeForConditionalGeneration`, `VideoLlama3ForConditionalGeneration`, `VideoLlama3VisionModel`, `VitsModel`

**Still breaking:**
- ❌ `AriaForConditionalGeneration`
- ❌ `AriaModel`
- ❌ `AudioFlamingo3ForConditionalGeneration`
- ❌ `AyaVisionForConditionalGeneration`
- ❌ `Cohere2VisionForConditionalGeneration`
- ❌ `DbrxForCausalLM`
- ❌ `DbrxModel`
- ❌ `DeepseekVLForConditionalGeneration`
- ... and 36 more

**Minimal repro** (`AriaForConditionalGeneration`):
```bash
# In ~/projects/oss-model-graph-break-corpus, with venv activated:
echo '["AriaForConditionalGeneration"]' > /tmp/repro.json
python3 sweep/run_sweep.py sweep \
  --models /tmp/repro.json \
  --modes eval \
  --compile-kwargs '{"fullgraph": true, "dynamic": true, "backend": "eager"}' \
  --setup-script sweep/configs/animesh-logging-suppress.py
```

Break reason: `Encountered graph break when attempting to trace CALL_FUNCTION_EX: a variadic function call, e.g. f(*args):`

**Comment by penguinwu** [comment]
**Status update — corrected classification (2026-04-28 evening)**

Earlier today's per-issue walk on the [Animesh fullgraph sweep](https://github.com/penguinwu/oss-model-graph-break-corpus/issues/70) classified several VL/multimodal models in this issue's affected list as `eager_error`. On further investigation we found that classification was masking the real status — the source-build venv (`torch-181552`) was missing `libnvrtc-builtins.so.12.4`, causing first-eager-call CUDA reduce kernel JIT to fail. That blocked the fullgraph compile from running; the sweep recorded `eager_error` instead of the real `graph_break` verdict.

After fixing the venv (commits [`50a3d3d`](https://github.com/penguinwu/oss-model-graph-break-corpus/commit/50a3d3d) + [`e82f93b`](https://github.com/penguinwu/oss-model-graph-break-corpus/commit/e82f93b) — `nvidia-cuda-*-cu12` install + sitecustomize.py preload + USE_LAPACK + timm install) and re-running just the affected models, the corrected status:

**No new fullgraph wins for this issue's affected models** — the data-dep output shape ops (nonzero, repeat_interleave, masked_select) still fire as expected.

**Confirmed graph_break under fullgraph (22 work items)** — these were `eager_error` in the morning sweep, now correctly classified as still-breaking (relevant input for this issue):

| Model | Modes |
|---|---|
| Glm46VForConditionalGeneration | eval, train |
| Glm4vForConditionalGeneration | eval, train |
| Glm4vMoeForConditionalGeneration | eval, train |
| GlmOcrForConditionalGeneration | eval, train |
| Mistral3ForConditionalGeneration | eval, train |
| PaddleOCRVLForConditionalGeneration | eval, train |
| Qwen2VLForConditionalGeneration | eval, train |
| Qwen3VLForConditionalGeneration | eval, train |
| Qwen3VLMoeForConditionalGeneration | eval, train |
| VideoLlama3ForConditionalGeneration | eval, train |
| VideoLlama3VisionModel | eval, train |

**Repro for any of the above:**
```bash
# eval (replace MODEL with one from the affected list)
python tools/reproduce.py --mode eval --device cuda <MODEL>
# fullgraph + dynamic config used by the sweep:
python sweep/run_sweep.py sweep --models '[{"name":"<MODEL>","source":"hf"}]' \
    --modes eval train --workers 1 --timeout 180 \
    --output-dir /tmp/repro_<MODEL> \
    --compile-kwargs '{"fullgraph": true, "dynamic": true, "backend": "eager"}'
```

Filed by Otter on Peng's behalf as part of corpus-side hygiene cleanup. A separate timm-cluster re-run is in flight (~14 more models that may also have updates relevant to this issue).

**Comment by penguinwu** [comment]
## Closing this issue — superseded by per-operator sub-issues

Per the umbrella-split policy (decided 2026-05-04): umbrella issues bundle multiple operator-specific root causes and never close, hiding actual progress. Splitting into per-operator sub-issues makes progress measurable.

## What this umbrella actually contained (sweep 2026-05-03)

| Count | Operator | Sub-issue |
|---|---|---|
| 573 breaks (59 models) | `aten.nonzero.default` | #117 |
| 62 breaks (26 models) | `aten.repeat_interleave.Tensor` | #118 |
| 20 breaks (1 model — HiDream) | `aten.bincount.default` (NEW this week) | #119 |
| 10 breaks (1 model — FastSpeech2) | `aten.masked_select` | #120 |
| 6 breaks (3 models) | `aten.unique` / `aten._unique2` | #121 |

(Note: `Tensor.item()` with `capture_scalar_outputs=False` was sometimes lumped into "data-dependent ops" but is a different config knob — tracked separately as #115.)

Closing as superseded. Future progress will be visible as individual sub-issues close.
